sqlglot/dialects/tsql.py

```diff
@@ -983,7 +983,9 @@ def setitem_sql(self, expression: exp.SetItem) -> str:
             return super().setitem_sql(expression)
 
         def boolean_sql(self, expression: exp.Boolean) -> str:
-            if type(expression.parent) in BIT_TYPES:
+            if type(expression.parent) in BIT_TYPES or isinstance(
+                expression.find_ancestor(exp.Values, exp.Select), exp.Values
+            ):
```

Comment:
This is probably ok, but I'm a little worried we may transform booleans in the context of automatically-generated conjunctions/disjunctions into 1s and 0s, leading to incorrect T-SQL syntax like `1 AND 0`.

I'm not sure if or how we can arrive at that state, but it may not be a bad idea to guard against it by checking if e.g. the immediate parent is a `Paren`.

Reply:
I'm not clear on exactly what query would trigger that.

It would only happen if somehow was a condition like `((TRUE) OR (FALSE))` somehow nested within a `VALUES(...)` statement, right? Since there are no `exp.Paren` objects anywhere within the AST of the queries this change was made for:

```
INSERT INTO tbl (foo, bar) VALUES (True, False)
```
and
```
SELECT * FROM (VALUES(True, False)) as t (foo, bar)
```

checking if the immediate parent of the `exp.Boolean` is an `exp.Paren` would always return False?

However if someone finds a way to trigger incorrect behaviour then i'll happily fix it since they will have an example we can add to the test suite